### DISCUSSION TITLE
Lack of Operations for InlineArray

### DISCUSSION THREAD

**Discussion by YT-GameMaker** [dotnet/runtime#95741] [General] [answered]
Static methods in `Array`, such as `Clear`, `Sort`, `IndexOf`, etc., cannot be used with inline arrays. Perhaps a mechanism is needed to reuse this code instead of rewriting a new set.
As of now, what is the best way to clear the entire inline array, using `array = default` or writing a loop?

**Comment by gfoidl** [comment, marked as answer]
`InlineArray` is a struct, so just "clear" it with `myInlineArray = default`.
Or access it as span and use `Span.Clear`-method.

For you other operator-wishes also just use span, e.g.
```c#
using System.Runtime.CompilerServices;

Foo foo = new();
foo[3] = 42;
foo[9] = 3;

Span<int> span = foo;
span.Sort();

for (int i = 0; i < span.Length; ++i)
{
    Console.WriteLine(span[i]);
}

[InlineArray(10)]
public struct Foo
{
    private int _value;
}
```
